pkg/textparse/openmetricsparse.go

```diff
@@ -232,13 +233,18 @@ func (p *OpenMetricsParser) Next() (Entry, error) {
 	p.exemplarVal = 0
 	p.hasExemplarTs = false
 
-	switch t := p.nextToken(); t {
+	t := p.nextToken()
+	defer func() { p.prev = t }()
+	switch t {
 	case tEofWord:
 		if t := p.nextToken(); t != tEOF {
 			return EntryInvalid, errors.New("unexpected data after # EOF")
 		}
 		return EntryInvalid, io.EOF
 	case tEOF:
+		if p.prev != tEofWord {
```

Comment:
nit: We tend to capitalize all acronyms on Go e.g `EOF` as described [here](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

Reply:
Except that this variable is generated by Golex -- and the link you provide says:

> Code generated by the protocol buffer compiler is exempt from this rule. Human-written code is held to a higher standard than machine-written code.

While this is not protocol buffer compiler, it is machine generated code.